src/components/navigation/navigation-item/navigation-item.component.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  Component,
+  Input,
+  ViewEncapsulation,
+  ContentChildren,
+  QueryList,
+  Renderer2,
+  ElementRef,
+  Optional,
+  SkipSelf,
+  AfterContentInit,
+  ContentChild,
+  Output,
+  AfterViewInit,
+  HostListener
+} from '@angular/core';
+import { Router, RouterLinkActive, NavigationEnd, RouterState, ActivatedRoute, UrlTree } from '@angular/router';
+import 'rxjs/add/operator/filter';
+
+@Component({
+    selector: '[ux-navigation-item]',
+    templateUrl: './navigation-item.component.html',
+    host: {
+        '[class.active]': 'active',
+        '[class.selected]': 'expanded',
+    },
+    encapsulation: ViewEncapsulation.None
+})
+export class NavigationItemComponent implements AfterViewInit {
+    @Input() header: string;
+    @Input() icon: string;
+    @Input() expanded: boolean = false;
+    @Input() link?: string;
+
+    @Input()
+    set level(value: number) {
+      this._level = value;
+    }
+
+    get level(): number {
+      return this._level;
+    }
+
+    @Output()
+    get active() {
+      if (this.link) {
+          return this._router.isActive(this.link, true);
+      } else {
+          return this._active;
+      }
+    }
+
+    private _level: number = 2;
+    private _active: boolean = false;
+
+    @ContentChildren(NavigationItemComponent, { descendants: true })
+    private _children: QueryList<NavigationItemComponent>;
+
+    get children(): NavigationItemComponent[] {
+      return this._children.filter(item => item !== this);
+    }    
+
+    constructor(
+        private _elementRef: ElementRef,
+        private _renderer: Renderer2,
+        @Optional()
+        @SkipSelf()
+        private _parent: NavigationItemComponent,
+        @Optional() private _router: Router
+    ) {
+        this.level = _parent ? _parent.level + 1 : 2;
+
+        if (this._router) {
+          _router.events.filter(event => event instanceof NavigationEnd).subscribe(event => {              
+            if (this.link && this._router.isActive(this.link, true)) {
+                this.select();
+            }
+
+            this.expanded = this.hasActiveLink(this.link);
+          });
+        }
+    }
+
+    ngAfterViewInit() {
+      let ol = this._elementRef.nativeElement.querySelector('.nav');

```

Comment:
Use `const` and also possibly rename this variable to give better indication about what we are searching for

Reply:
Renamed variable in ngAfterViewInit() and declared as const.